src/stickApi.pas

```diff
@@ -2,56 +2,225 @@
+
+procedure printTop(const args: array of const);
+var
+  i: Integer;
+  route, output, line, nev, pont: string;
+  response: TApiResponse;
+begin
+  route := VariantUtils.VarRecToStr(args[0]);
+
+  try
+    response := TApi.GET(route);
+
+    if response.success then
+    begin
+      for i := 0 to 9 do
+      begin
+        nev := response.data.getString(['data', i, 'nev']);
+        if length(nev) = 0 then nev := '-';
+
+        pont := response.data.getString(['data', i, 'pont']);
+        if length(pont) = 0 then pont := '';
+
+        line := intToStr(i + 1) + '. ' + nev + ' ' + pont;
+        output := output + NL + line;
+       end;
+       scriptsHandler.evalscriptline('display ' + output);
+    end;
+
+  except
+    sentryModule.addBreadcrumb(makeBreadcrumb('printTop failed on ' + route));
+    //TODO: reportError(E, msg) helyette majd ha nem egy file lesz az output
+  end;
+
+end;
+
+procedure printRank(const args: array of const);
+var
+  route, output, username, mode: string;
+  response: TApiResponse;
+begin
+  try
+    username := VariantUtils.VarRecToStr(args[0]);
+    mode := VariantUtils.VarRecToStr(args[1]);
+    route := 'rank&nev=' + username + '&type=' + mode;
+
+    response := TApi.GET(route);
+
+    if response.success then
+    begin
+      output := response.data.getString(['data', 'rank']);
+
+      scriptsHandler.evalscriptline('display ' + output);
+    end;
+
+  except
+    on E:Exception do
+    begin
+      sentryModule.addBreadcrumb(makeBreadcrumb('printRank failed on ' + route));
+      //TODO: reportError(E, msg) helyette majd ha nem egy file lesz az output
+    end;
+  end;
+end;
+
+procedure printKoth(const args: array of const);
+var               
+  i: Integer;
+  output, line, nev, pont, mode: string;
+  response: TApiResponse;
+begin
+  try
+    mode := VariantUtils.VarRecToStr(args[0]);
+    response := TApi.GET(mode);
+
+    if response.success then
+    begin
+      for i := 0 to 9 do
+      begin
+        nev := response.data.getString(['data', i, 'nev']);
+        if length(nev) = 0 then nev := '-';
+
+        pont := response.data.getString(['data', i, 'pont']);
+        if length(pont) = 0 then pont := '';
+
+        line := intToStr(i + 1) + '. ' + nev + ' ' + pont;
+        output := output + NL + line;
+       end;
+       
+       scriptsHandler.evalscriptline('display ' + output);
+    end;
+  except
+    on E:Exception do
+    begin
+      sentryModule.addBreadcrumb(makeBreadcrumb('printKoth failed on ' + mode));
+      //TODO: reportError(E, msg) helyette majd ha nem egy file lesz az output
+    end;
+  end;
+end;
+
+procedure sendBotKills(const args: array of const);
+var
+  route, kills: string;
+  response: TApiResponse;
+  postData: TIdMultiPartFormDataStream;
+begin
+  route := 'botkill';
+
+  try
+    kills := VariantUtils.VarRecToStr(args[0]);
+
+    postData := TIdMultiPartFormDataStream.Create;
+    postData.AddFormField('form[nev]', KillMeUtils.unhungaryify(multisc.nev));
```

Comment:
Ez nem a legjobb, de így lekezeli a speckó karaktereket is és nem kell szerver oldalon encodinggal hülyéskedni.